src/main/java/com/minecolonies/coremod/commands/GetColonyAndCitizen.java

```diff
@@ -0,0 +1,216 @@
+    private static final String UNKNOWN_ERROR           = "Unknown Error!";
+    private static final String NOT_FOUND               = "%s not found!";
+    private static final String NO_COLONY               = "You haven't got a colony!";
+    private static final int    ONLY_NUMBERS_CODE       = -1;
+    private static final int    TOO_MANY_ARGUMENTS_CODE = -2;
+    private static final int    UNKNOWN_ERROR_CODE      = -3;
+    private static final int    NOT_FOUND_CODE          = -4;
+    private static final int    NO_COLONY_CODE          = -5;
+
+    private GetColonyAndCitizen()
+    {
+        throw new IllegalAccessError("Utility Class");
+    }
+
+    /**
+     * Getting the colony ID.
+     */
+    public static int getColonyId(@NotNull final ICommandSender sender, @NotNull final String... args)
+    {
+        int colonyId;
+        final UUID mayorID = sender.getCommandSenderEntity().getUniqueID();
+        if (args.length == 2 || args.length == 4)
+        {
+            try
+            {
+                colonyId = Integer.parseInt(args[0]);
+            }
+            catch (NumberFormatException e)
+            {
+                colonyId = ONLY_NUMBERS_CODE;
+            }
+        }
+        else if (args.length == 1 || args.length == 3)
+        {
+            if (ColonyManager.getIColonyByOwner(sender.getEntityWorld(), mayorID) == null)
+            {
+                colonyId = NO_COLONY_CODE;
+            }
+            else
+            {
+                colonyId = ColonyManager.getIColonyByOwner(sender.getEntityWorld(), mayorID).getID();
+            }
+        }
+        else if (args.length >= 5)
+        {
+            colonyId = TOO_MANY_ARGUMENTS_CODE;
+        }
+        else
+        {
+            colonyId = UNKNOWN_ERROR_CODE;
+        }
+        if (colonyId >= 0)
+        {
+            if (ColonyManager.getColony(colonyId) == null)
+            {
+                colonyId = NOT_FOUND_CODE;
+            }
+        }
+
+
+        return colonyId;
+    }
+
+    /**
+     * Getting the citizen ID.
+     * @param colonyId
+     * @param args
+     * @return
+     */
+    public static int getCitizenId(@NotNull final int colonyId, @NotNull final String... args)
+    {
+        int citizenId;
+        String citizenName;
+        citizenId = NOT_FOUND_CODE;
+        if (args.length == 2)
+        {
+            try
+            {
+                citizenId = Integer.parseInt(args[1]);
+            }
+            catch (NumberFormatException e)
+            {
+                citizenId = ONLY_NUMBERS_CODE;
+            }
+        }
+        else if (args.length == 1)
+        {
+            try
+            {
+                citizenId = Integer.parseInt(args[0]);
+            }
+            catch (NumberFormatException e)
+            {
+                citizenId = ONLY_NUMBERS_CODE;
+            }
+        }
+        else if (args.length == 3 && colonyId >= 0)
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Assign this magic number 3 to a well-named constant, and use the constant instead. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://home.kk-sc.de:9000/coding_rules#rule_key=squid%3AS109)